enhancements/baremetal/enable-baremetal-on-other-platforms.md

```diff
@@ -0,0 +1,198 @@
+manage baremetal hosts from clusters without requiring the cluster to be on baremetal
+would be beneficial to customers.
+
+## Motivation
+
+An initial driver of this feature are the centralized host management use cases
+in edge topologies, which without this feature, is restricted to having the
+central OpenShift cluster deployed on baremetal.
+
+See:
+- https://github.com/openshift/enhancements/blob/master/enhancements/installer/agent-based-installation-in-hive.md
+- https://github.com/openshift/assisted-service/tree/master/docs/hive-integration
+
+### Goals
+
+The specific goals of this proposal are to:
+
+Support the centralized host management use case by partially enabling Baremetal Host API
+on the following platforms:
+- None
+- OpenStack
+- vSphere
+
+We will be successful when:
+
+Centralized host management can deploy clusters when running on the above platforms.
+
+### Non-Goals
+
+Allow Baremetal Host API to be fully enabled on all platforms.
+
+Allow Machine API integration across platform types.
+- The centralized host management flow currently interacts directly via this API
+  without any Machine API integration.
+- This means further work will be required via a future enhancement to enable the
+  single cluster case where a combination of e.g VM controlplane and Baremetal
+  workers is desired.
+
+## Proposal
+
+BMO (baremetal-operator) provides the Baremetal Host API, it in turn is configured
+and managed by CBO (cluster-baremetal-operator).
+
+CBO reads the Provisioning CR that is created by the installer on baremetal platforms
+and uses that to configure and deploy BMO. In the case of non-baremetal platforms
+the user (or automation) will need to define the Provisioning CR.
+
+Currently CBO checks the platform and if it is not baremetal it will be in a "disabled" state i.e. it will
+1. set status.conditions Disabled=true and
+2. not read or process the Provisioning CR and thus not deploy baremetal-operator.
+
+This proposal is to allow CBO to be enabled on the following platforms only:
+- Baremetal (current)
+- None
+- OpenStack
+- vSphere.
+
+Note: The Hypershift use case will be explicitly disallowed by disabling CBO when
+Infrastructure.Status.ControlPlaneTopology == "External".
+
+Further (to restrict the testing matrix) the allowed configuration options
+of the Provisioning CR will be restricted to exactly those required by centralized host management.
+
+*Only spec.provisioningNetwork=Disabled mode will be accepted in the Provisioning CR.*
+
+If any other provisioningNetwork mode is set, the CBO webhook will refuse the change
+in the usual case, but if defined before upgrading the operator, the Reconcile loop
+must always validate the Provisioning CR.
+
+Note:
+
+1. when the Provisioning CR is set to provisioningNetwork=Disabled mode, worker
+nodes would be booted via virtual media. This removes the requirement for the
+Provisioning Network which can be expected to be available only in Baremetal platform types.
```

Comment:
IIRC, the `ProvisioningOSDownloadURL`, is required. Should/Will it till be required when running on other platforms? 

From the [field's docstring](https://github.com/openshift/cluster-baremetal-operator/blob/cae7b845e8eae10c12e4540e0d83c2bc972d8bb5/api/v1alpha1/provisioning_types.go#L124-L126):

> 	// ProvisioningOSDownloadURL is the location from which the OS
>	// Image used to boot baremetal host machines can be downloaded
>	// by the metal3 cluster.

Reply:
Hold on, my comment is about making `ProvisioningOSDownloadURL` not-required as well, not about `provisioningNetwork` :stuck_out_tongue_closed_eyes:

Reply:
> The ZTP flow however is different/special, in that it only currently deploys ISO images generated by assisted-service, thus it may well be possible to omit ProvisioningOSDownloadURL in that case, provided CBO has sufficient conditionals to just not deploy the download/cache containers in that case?

Chatted with @asalkeld yesterday and it looks like it may be possible to remove the validation that makes this data required. So far, we've been using whatever we find there or whatever matches the version of the cluster where `metal3` is running. However, I think it'd be better if we could not fill in this field with semi-random data.

Thanks for the comment and confirming. I'm unsure on whether there are enough validations in CBO to prevent deployments when this data is not provided but it would be great if there were.